packages/edit-post/src/components/options-modal/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { PostTaxonomies } from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { EnablePublishSidebarOption, EnableTipsOption, EnablePanelOption } from './options';
+
+const MODAL_NAME = 'edit-post/options';
+
+const Section = ( { title, children } ) => (
+	<section className="edit-post-options__section">
+		<h2 className="edit-post-options__section-title">{ title }</h2>
+		{ children }
+	</section>
+);
+
+export function OptionsModal( { isModalActive, toggleModal } ) {
+	if ( ! isModalActive ) {
+		return null;
+	}
+
+	return (
+		<Modal
+			className="edit-post-options"
+			title={ <span className="edit-post-options__title">{ __( 'Options' ) }</span> }
+			closeLabel={ __( 'Close' ) }
+			onRequestClose={ toggleModal }
+		>
+			<Section title={ __( 'General' ) }>
+				<EnablePublishSidebarOption label={ __( 'Enable Pre-publish Checks' ) } />
+				<EnableTipsOption label={ __( 'Enable Tips' ) } />
+			</Section>
+			<Section title={ __( 'Document Panels' ) }>
+				<EnablePanelOption panelName="post-status" label={ __( 'Status & Visibility' ) } />
+				<PostTaxonomies
+					taxonomyWrapper={ ( content, taxonomy ) => (
+						<EnablePanelOption
+							label={ get( taxonomy, [ 'labels', 'menu_name' ] ) }
+							panelName={ `taxonomy-panel-${ taxonomy.slug }` }
+						/>
+					) }
+				/>
+				<EnablePanelOption label={ __( 'Featured Image' ) } panelName="featured-image" />
```

Comment:
🤔🤔🤔 

OK, so currently we have:

| Component | Identifier |
| --- | --- |
| `PostStatus` | `post-status` |
| `PostTaxonomies` | `taxonomy-panel-*` |
| `FeaturedImage` | `featured-image` |
| `PostExcerpt` | `post-excerpt` |
| `DiscussionPanel` | `discussion-panel` |
| `PageAttributes` | `page-attributes` |

Here's my proposal:

| Component | Identifier |
| --- | --- |
| `StatusPanel` | `status` |
| `PostTaxonomiesPanel` | `post-taxonomy-*` |
| `FeaturedImagePanel` | `featured-image` |
| `PostExcerptPanel` | `post-excerpt` |
| `DiscussionPanel` | `discussion` |
| `PageAttributesPanel` | `page-attributes` |

These names ensure:

- Identifiers consistently don't contain the word `panel`.
- Component names end with `Panel` which properly describes what the UI component actually _is_.
- Panels that only apply to posts begin with `post-`.
- Panels that only apply to pages begin with `page-`.

What do you think? Is it worth changing the names? It's tricky to do this and ensure backwards compatibility with the stored `preferences`.

Reply:
When you check closely code of the panels, it seems like all of them but `PageAttributesPanel` should be prefixed with `Post`. I also agree that all of components should end with `Panel`. My vote would be for:

| Component | Identifier |
| --- | --- |
| `PostStatusPanel` | `post-status` |
| `PostTaxonomiesPanel` | `post-taxonomy-*` |
| `PostFeaturedImagePanel` | `post-featured-image` |
| `PostExcerptPanel` | `post-excerpt` |
| `PostDiscussionPanel` | `post-discussion` |
| `PageAttributesPanel` | `page-attributes` |

What do you think?

Reply:
By the way, this can be done in a seperate PR, this isn't a blocker at all as it existed before. It would be nice to have it standardized as we will be referencing them in the modal dialog. By they way, do you think we should put those values in constants or would it be overkill?

Reply:
> By they way, do you think we should put those values in constants or would it be overkill?

It's probably overkill. We don't use constants for other identifiers e.g. block names and modal names.

> When you check closely code of the panels, it seems like all of them but PageAttributesPanel should be prefixed with Post.

I disagree. I think:

- Panels that appear on both posts and pages (e.g. Discussion) should have no prefix
- Panels that appear on posts but not pages (e.g. `PostExcept`) should have a `Post` prefix
- Panels that appear on pages but not posts (e.g. `PageAttributes`) should have a `Page` prefix

Reply:
> Panels that appear on both posts and pages (e.g. Discussion) should have no prefix

I like your reasoning. I think it's more complex in the WordPress world because everything is a post both `Post` and `Page` which makes it all even more confusing.

I'm wondering how Classic Editor handles it.